### PR TITLE
Simplify annotations in `_brute_force.py`

### PR DESCRIPTION
**Pull request by eukaryo**
## Motivation

Related to https://github.com/optuna/optuna/issues/4508 ,

### CODE REVIEW

File: optuna/samplers/_brute_force.py [hunk 1]
```diff
@@ -3,12 +3,8 @@
 from dataclasses import dataclass
 import decimal
 from typing import Any
-from typing import Dict
 from typing import Iterable
-from typing import List
-from typing import Optional
 from typing import Sequence
```
Comment (nabenabe0928) [inline optuna/samplers/_brute_force.py:7]: `Iterable` and `Sequence` should be imported from `collections.abc`.
https://peps.python.org/pep-0585/
```suggestion
from collections.abc import Iterable
from collections.abc import Sequence
```

**Review by nabenabe0928 — COMMENTED** [review]
I left one comment!

**Comment by nabenabe0928** [comment]
@not522 Could you review this PR?

File: optuna/samplers/_brute_force.py (hunk 1 above)
Comment (eukaryo) [reply optuna/samplers/_brute_force.py]: Thank you for your comment! I revised the code.

**Review by nabenabe0928 — APPROVED** [review]
LGTM!

**Review by not522 — APPROVED** [review]
LGTM

**Pull request merged**